services/_shared/adapters/pg_install_store.py

```
@@ -286,14 +290,26 @@ def set_repo_config(
     # caller ever passed a query-string value through.
+    # Voice is an exception: it's a Literal["caveman", "sage"] (or None).
+    _ALLOWED_STRING_FLAGS = {"elder_voice"}
     non_bool = {
         flag: value for flag, value in persona_flags.items()
         if value is not None and not isinstance(value, bool)
+        and flag not in _ALLOWED_STRING_FLAGS
     }
     if non_bool:
         raise TypeError(
             f"set_repo_config() persona flags must be bool or None; got {non_bool!r}"
         )
```

Ruff (0.15.20)

[warning] Avoid specifying long messages outside the exception class

(TRY003)